en/docs/apis/advanced-topics/connect-to-an-external-identity-provider.md

```diff
@@ -0,0 +1,146 @@
+        You can update this default value via the Okta authorization server by navigating to the following path:<br>
+        Security → API → Authorization Server → Your Authorization Server → Settings → Audience<br><br>
+        <b>Microsoft:</b><br>The identifier that identifies the intended recipient of the token.
+This value should be the application ID URI of the application in Azure Active Directory (Azure AD) for which the token is requested.
+        <br><br>
+        <b>Auth0:</b><br>The unique API identifier to use as the audience parameter in the authorization call.<br>
+         You can update this value in the Auth0 server by navigating to the following path:<br>
+         Applications → APIs → Your API → General Settings → Identifier<br>
+    </td>
+    </tr>
+    <tr class="even">
+    <td>**Well-Known URL**</td>
+    <td><p>The OpenID Connect discovery endpoint URL. This returns the metadata related to the OpenID Provider's configuration.</p>
+    </td>
+    </tr>
+    </table>
+
+     ![Add an identity provider](../../assets/img/apis/advanced-topics/external-identity-providers/add-an-identity-provider.png){.cInlineImage-half} 
+
+     When you specify the **Well-Known URL**, the values for the **Issuer**, **Token Endpoint**, and **JWKS Endpoint** fields are filled in automatically in the next screen. Depending on your requirement, you can either keep the values as it is or manually change the values. The following table describes each field in detail:
+
+    <table>
+    <tr class="header">
+    <th><b>Field</b></th>
+    <th><b>Description</b></th>
+    </tr>
+    <tr class="odd">
+    <td>**Issuer**</td>
+    <td>The issuer identifier of the IdP, which is in the `iss` claim of the JWT issued.</td>
+    </tr>
+    <tr class="even">
+    <td>**Token Endpoint**</td>
+    <td>The token endpoint URL of the IdP from where the OAuth client can get an access token.</td>
+    </tr>
+    <tr class="odd">
+    <td>**JWKS Endpoint**</td>
+    <td>The URL that returns the JSON web key (JWK) set of the IdP. This returns a collection of JSON Web Keys used to verify the signature of the JWT tokens.
+    </td>
+    </tr>
+    </table>
+
+     ![Identity provider details](../../assets/img/apis/advanced-topics/external-identity-providers/identity-provider-details.png){.cInlineImage-half}
+
+8.  Click **Add**. This adds Okta IdP to the list of existing identity providers in your organization.
+
+Now that you have added an external IdP, you can go ahead and obtain access tokens from that IdP to exchange and get Choreo access tokens to invoke APIs.
+
+
+## Obtain an access token from an external identity provider
+
+Follow this procedure to obtain an access token from an external IdP and exchange it with a Choreo token within your organization:
+
+1. Sign in to the Choreo API Developer Portal at [https://devportal.choreo.dev/](https://devportal.choreo.dev/).
```

Comment:
The numbering is in roman numbers. Shouldn't it be arabic?

Reply:
It is in arabic when rendered. Not sure why it is in roman numerals when viewing the file.
<img width="1253" alt="numbering" src="https://user-images.githubusercontent.com/9528121/131962599-5336c057-ee49-460f-8fd9-0ad0889914e0.png">

Reply:
Ah then fine.